processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1098,6 +1098,11 @@ public static File makeTimeFile(File dir, ByteOrder order)
 
   public static File makeMetricFile(File dir, String metricName, ByteOrder order)
   {
-    return new File(dir, String.format("met_%s_%s.drd", metricName, order));
+    return new File(dir, sanitizeFileName(String.format("met_%s_%s.drd", metricName, order)));
+  }
+
+  public static String sanitizeFileName(String fileName)
+  {
+    return fileName.replace(File.separator, "↗");
```

Comment:
that's fine too, my only objective is to preserve dimension names as much as possible to keep debugging easy.
@drcrallen actually pointed out file system limits are in bytes, and we don't know what encoding the file system might use, so we'll have to take that into account.